### PR TITLE
[Compute] BREAKING CHANGE: `az vm/vmss create`: Support Trusted Launch as default deployment option

### PR DESCRIPTION
**Pull request by yanzhudd**
**Related command**

az vm/vmss create

**Description**

Merge changes from:
close: #27636 
close: #27722
close: #27635

Feature requests:
close: #23450 
close: #23476 
close: #23440 

**Testing Guide**

**History Notes**

[Compute] BREAKING CHANGE: `az vm/vmss create`: Support Trusted Launch as default deployment option when creating vm/vmss with marketplace image
[Compute] BREAKING CHANGE: `az vm/vmss create`: Support Trusted Launch as default deployment option when creating vm/vmss from existing disk or image
[Compute] BREAKING CHANGE: `az vm/vmss create`: Support Trusted Launch as default deployment option when creating vm/vmss with minimal inputs

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

**Review by necusjz — DISMISSED** [review]

**Review by zhoxing-ms — DISMISSED** [review]

File: src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py [hunk 1]
```diff
@@ -602,6 +602,32 @@ def trusted_launch_warning_log(namespace, generation_version, features):
...
+        logger.warning(UPGRADE_SECURITY_HINT)
+
+
+def validate_image_trusted_launch(namespace):
+    from ._constants import UPGRADE_SECURITY_HINT
+
+    # set securityType to Standard by default if no inputs by end user
+    if namespace.security_type is None:
+        namespace.security_type = 'Standard'
+    if namespace.security_type.lower() != 'trustedlaunch':
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py:627]: Just a small question, why sometime the judgement logic is ` if security_type == 'TrustedLaunch'`?

Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py:627]: This param is defined with `arg_type=get_enum_type` in _params.py, and it is converted to `TrustedLaunch` if the input is `trustedlaunch`. Hence `security_type == 'TrustedLaunch'` and `namespace.security_type.lower() != 'trustedlaunch'` both work.
The difference is caused by the different developing style between Taoxu and me hahaha
Will keep consistence.

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py:627]: OK, sounds good

**Review by zhoxing-ms — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Pull request merged**